pkg/controller/podgc/gc_controller_test.go

```diff
@@ -136,29 +128,30 @@ func TestGCTerminated(t *testing.T) {
 		},
 	}
 
-	for i, test := range testCases {
+	for _, test := range testCases {
 		t.Run(test.name, func(t *testing.T) {
-			client := fake.NewSimpleClientset(&v1.NodeList{Items: []v1.Node{*testutil.NewNode("node")}})
-			gcc, podInformer, _ := NewFromClient(client, test.threshold)
-
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PodDisruptionConditions, test.enablePodDisruptionConditions)()
 			creationTime := time.Unix(0, 0)
+			nodes := []*v1.Node{testutil.NewNode("node")}
+
+			pods := make([]*v1.Pod, 0, len(test.pods))
 			for _, pod := range test.pods {
 				creationTime = creationTime.Add(1 * time.Hour)
-				podInformer.Informer().GetStore().Add(&v1.Pod{
+				pods = append(pods, &v1.Pod{
 					ObjectMeta: metav1.ObjectMeta{Name: pod.name, CreationTimestamp: metav1.Time{Time: creationTime}},
 					Status:     v1.PodStatus{Phase: pod.phase},
 					Spec:       v1.PodSpec{NodeName: "node"},
 				})
 			}
+			client := setupNewSimpleClient(nodes, pods)
+			gcc, podInformer, _ := NewFromClient(client, test.threshold)
+			for _, pod := range pods {
+				podInformer.Informer().GetStore().Add(pod)
+			}
```

Comment:
this seems to differ from previous implementation, the client was only initialized with one node, now it seems to have pods too. 

However, I don't know how the clientset and the infomer initialized from the clientset works, is that the infomer automatically store the clientset initialized values?
if that is true, do we need to addd pods again to the informer storage?

Reply:
Yes, they need to be added when constructing the client so that the patch operation does not fail with "not found" error. Actually, previously, the delete operation was failing with the NotFound error in the tests, but the tests didn't care - they only tested if the Delete action was received. Now, when the feature is enabled, I need the pods to be in the client in order to reach the delete operation. 

As for the need to add them to the pod informer - this is also needed (maybe it could be avoided with some refactoring), because if I don't add them to the informer then this list is empty: https://github.com/kubernetes/kubernetes/blob/719f3cf8da49d94da73b9ada1760e6f94e9834dd/pkg/controller/podgc/gc_controller.go#L111